tensorflow_addons/layers/python/maxout_test.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for Maxout layer."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+
+from tensorflow.python.keras import testing_utils
+from tensorflow.python.keras.utils import generic_utils
+from tensorflow.python.platform import test
+from tensorflow_addons.layers.python.maxout import Maxout
+
+
+class MaxOutTest(test.TestCase):
+    def test_simple(self):
+        # TODO: more simple way to deserialize the layers in addons.
+        with generic_utils.custom_object_scope({'Maxout': Maxout}):
```

Comment:
tf.keras cannot recognize custom layers, I think there are at least two ways to mitigate the problem:
1. We inject our custom layers into `_GLOBAL_CUSTOM_OBJECTS` of keras (or use `CustomObjectScope.__enter__()`) when importing. Users don't need do anything.
2. We create a `custom_objects` for addons.layers, and ask users to pass them to `Model.from_config(config, custom_objects)` explicitly.

Reply:
Do we need to create an issue in addons to discuss the problem?

Reply:
At first glance I lean toward option 1. Perfectly okay with an issue to discuss it though. I started #23 so we could formalize this requirement once a decision is made